### PR TITLE
mgr/dashboard: Fix RGW user/bucket quota issues

### PR DESCRIPTION
**Pull request by votdev**
* RGW user/bucket "max. size" should be hidden when "user/bucket quota" is not enabled
* Only execute validators if necessary
* Do not show "Maximum size" and "Maximum objects" values in details if quota is disabled

Fixes: https://tracker.ceph.com/issues/39964

Signed-off-by: Volker Theile <vtheile@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [x] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please.

**Comment by votdev** [comment]
> I've set the "Max. objects":
> 
> Screenshot from 2019-05-23 10-48-45
> 
> Then I've unselected the "Enabled" checkbox:
> 
> Screenshot from 2019-05-23 10-35-54
> 
> But after submission, when I look into the details, I see the "Max. objects" value:
> 
> Screenshot from 2019-05-23 10-35-20
> 
> When I edit the user I don't see it:
> 
> Screenshot from 2019-05-23 10-35-54
> 
> Is this an expected behavior? Maybe we should not submit those values when "Enabled" is not selected.

Good spotting, i think we should hide the other values when ``User quota`` is ``disabled``, or display a ``n/a```or ``-`` in this case.

I do not want to change the behaviour of the form dialog which is currently that when the quota is disabled, the manually entered values are still transferred to the RGW API because some values MUST be submitted. I do not want to override/manipulate the user values here to be able to show them again when the user decides to enable quota and manual ``Maximum size`` values again.

**Comment by votdev** [comment]
> > I've set the "Max. objects":
> > Screenshot from 2019-05-23 10-48-45
> > Then I've unselected the "Enabled" checkbox:
> > Screenshot from 2019-05-23 10-35-54
> > But after submission, when I look into the details, I see the "Max. objects" value:
> > Screenshot from 2019-05-23 10-35-20
> > When I edit the user I don't see it:
> > Screenshot from 2019-05-23 10-35-54
> > Is this an expected behavior? Maybe we should not submit those values when "Enabled" is not selected.
> 
> Good spotting, i think we should hide the other values when `User quota` is `disabled`, or display a `n/a` or `-` in this case.
> 
> I do not want to change the behaviour of the form dialog which is currently that when the quota is disabled, the manually entered values are still transferred to the RGW API because some values MUST be submitted. I do not want to override/manipulate the user values here to be able to show them again when the user decides to enable quota and manual `Maximum size` values again.

When quota is enabled:
Auswahl_001

When quota is disabled:
Auswahl_002

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-details/rgw-user-details.component.html [hunk 1]
```diff
@@ -68,21 +68,27 @@
             <tr>
               <td i18n
                   class="bold col-sm-1">Maximum size</td>
-              <td *ngIf="user.user_quota.max_size <= -1"
+              <td *ngIf="!user.user_quota.enabled"
+                  i18n
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-details/rgw-user-details.component.html]: Is it necessary to translate this?

Comment (votdev) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-details/rgw-user-details.component.html]: No :-)

**Comment by votdev** [comment]
jenkins test make check arm64

**Pull request merged**